docs/contribute/module.md

```diff
@@ -11,18 +12,89 @@ Each experimentalist, experiment runner or theorist is a "child" package based o
+For example, if you are implementing an experiment sampler, you may rename and modify the 
+``tests/test_experimentalist_sampler_example.py``.
+
+*Note: Tests are required if you wish that your feature becomes part of the main 
+[autora](https://github.com/AutoResearch/autora) package. However, regardless of whether you choose to implement tests, 
+you will still be able to install your package separately, in addition to autora.* 
+
+### Step 3 (Optional): Add Documentation
+
+It is highly encouraged that you add documentation of your package in your `docs/index.md`. You can also add new pages 
+in the `docs` folder. Update the `mkdocs.yml` file to reflect structure of the documentation. For example, you can add 
+new pages or delete pages that you deleted from the `docs` folder.
+
+You are also encouraged to describe basic usage of your theorist in the 
+python notebook ``Basic Usage.ipynb`` in the `docs` folder. Finally you can outline the basic setup of your theorist in 
+the `docs/quickstart.md` file.
+
+*Note: Docmentation is required if you wish that your feature becomes part of the main 
+[autora](https://github.com/AutoResearch/autora) package. However, regardless of whether you choose to write
+documentation, you will still be able to install your package separately, in addition to autora.*
+
+### Step 4: Add Dependencies
+
+In pyproject.toml add the new dependencies under `dependencies`
+
+Install the added dependencies
+```shell
+pip install -e ".[dev]"
+```
+
+## Publishing your module
+
+There are several ways to publish your package, depending on how you set up your repository.
+
+- If you used the **cookiecutter template**, and uploaded your repository to 
+github.com, then you can use Github Actions to automatically publish your package to PyPI. 
+
+- If you used the **unguided template**, or you want to manually publish your package, you can follow [step 7 in this guide](https://github.com/AutoResearch/autora-template).
+
+Once you've published your module on PyPI, you should take some time to celebrate and announce your contribution in the 
+[AutoRA forum](https://github.com/orgs/AutoResearch/discussions/categories/module-announcements).
+
+## Including your module into the autora parent package
+
 Once your package is working, and you've published it on PyPI, you can **make a pull request** on 
 [`autora`](https://github.com/autoresearch/autora) to have it vetted and added to the "parent" package.
-
 The following demonstrates how to add a package published under autora-theorist-example in PyPI in the GitHub 
 repository example-contributor/contributor-theorist
 
-## Creating a new child package
+!!! success
+    In order for your package to be included in the parent package, it must
+    - include basic documentation in ``docs/index.md``
+    - include a basic python notebook exposing how to use the module in ``docs/Basic Usage.ipynb``
+    - include basic tests in ``tests/``
+    - be published on PyPI
```

Comment:
i think in this case it makes sense, but just to be sure, limiting our consideration to PyPI?

Reply:
Great point! should be both conda or PyPI